feat(profiling): add span component

JonasBa opened the pull request:
Adds the canvas and relevant boilerplate for the spans timeline component. Feature flagged behind `organizations:profiling-flamechart-spans`.

JonasBa committed:
feat(profiling): add span tree

JonasBa committed:
feat(profiling): use rawspantype

JonasBa committed:
feat(profiling): add iterator to spantre

JonasBa committed:
feat(profiling): add spans view component

JonasBa committed:
feat(profiling): feature flag spans component

JonasBa commented on static/app/utils/profiling/flamegraph/flamegraphTheme.tsx:
```diff
@@ -47,14 +47,7 @@ export interface FlamegraphTheme {
     LABEL_FONT_COLOR: string;
     MINIMAP_POSITION_OVERLAY_BORDER_COLOR: string;
     MINIMAP_POSITION_OVERLAY_COLOR: string;
-    REQUEST_2XX_RESPONSE: string;
```
We copied these from specto at the time, but they are not used anywhere so 🔪

Zylphrex commented on static/app/components/profiling/flamegraph/flamegraphLayout.tsx:
```diff
@@ -116,17 +122,20 @@ const FlamegraphGrid = styled('div')<{
       : layout === 'table right'
       ? `
         'minimap    frame-stack'
+        'spans    frame-stack'
```
```suggestion
        'spans     frame-stack'
```

JonasBa committed:
Update static/app/components/profiling/flamegraph/flamegraphLayout.tsx

Co-authored-by: Tony Xiao <txiao@sentry.io>

JonasBa committed:
feat(profiling): wire up the renderer (#42536)

This PR adds and wires up the actual span renderer (for now just use 2D
context, we'll 99% likely be fine as span count should be < 1000).

As it stands, nothing will actually render because the durations are not
normalized and time axis is off - I will be tackling these in separate
smaller PRs.

JonasBa committed:
feat(profiling): add spanChart component (#42492)

Adds the SpanChart component - this is the component responsible for
outputting chart nodes that need to be rendered. The model pretty much
follows the profile approach where we generate a ds (in this case
SpanTree) and then create a component that is responsible for outputting
and mapping that ds to the actual space that we will render it in.

Zylphrex reviewed (approved):
🚢

JonasBa committed:
fix(eslint): sort imports and fix double import
